content/posts/sixtyfour.md

```diff
@@ -0,0 +1,199 @@
+
+```r
+Sys.setenv(
+  AWS_ACCESS_KEY_ID = "your key",
+  AWS_SECRET_ACCESS_KEY = "your secret",
+  AWS_REGION = "us-west-2"
+)
+```
+
+Create a user: creating a user is not just a single API call to AWS. To have a real user setup that can access AWS they need to have a username in an AWS account, then have some policies attached and access keys before they can get any information from the AWS API. `six_user_create()` not only creates the user but copies an email template with the keys for the user to make it easy to send the information to the person behind the new user name.
+
+```r
+user <- random_user()
+six_user_create(user)
+```
+
+Create a bucket
+
+```r
+bucket <- random_bucket()
+aws_bucket_create(bucket)
+```
+
+Add the user to the bucket so they can access it with read permissions: adding a user to a bucket takes a few steps: create a bucket policy if missing, then attach the policy to the user. In addition, we've simplified the permissions to just "read" and "write", handling the necessary AWS policies behind the scenes to give a user read or write access. Of course you can drop down to lower level `aws_*` functions if you need more control.
+
+```r
+six_bucket_add_user(
+  bucket = bucket,
+  username = user,
+  permissions = "read"
+)
+```
+
+## Sensitive information
+
+When interacting with AWS there's many pieces of information that you likely do not want shown to other people. These include:
+
+- Access keys
+- Secret keys
+- AWS Account ID
+- AWS ARN's that include sensitive information (like Account ID)
+- AWS Region
+- and surely more
+
+We used a few different approaches to safeguard users (including us as we develop the package):
+
+- Many packages allow credentials to be passed in as arguments to functions. We don't allow this because it would make it easy for users to pass raw strings in to functions that are sensitive. `paws` already handled finding credentials and so we rely on it.
+- Provide the function `aws_configure()` to let users redact secrets in outputs from the AWS API - and the  `with_redacted` function allows you to temporarily redact sensitive information from the output of a code block (like a `with` context manager in Python). What does this look like? Here's an example:
+
+```r
+with_redacted(
+  new_user <- six_user_create(random_user())
+)
+#> ℹ Added policy UserInfo to BarreledRunoff
+#> ✔ Key pair created for BarreledRunoff
+#> ℹ AccessKeyId: *****
+#> ℹ SecretAccessKey: *****
+#> Email template copied to your clipboard
+```
+
+If we didn't use `with_redacted`, the output would show the access key ID and the secret. You haven't lost the actual key and secret though assuming you assigned it to a variable (here `new_user`) - even if you forget to assign to a variable you can always do `.Last.value`.
+
+In addition, for `six_user_create` we copy an email template to your clipboard for sharing the new AWS user with the real person it should be associated with.
```

Comment:
You also mention the six_user_create email template on line 47.

Reply:
thanks, removed